terraform_compliance/common/helper.py

```diff
@@ -565,17 +577,12 @@ def merge_dicts(source, target):
 
         # add remaining values of lower priority to higher priority
         ignore non dict values
-
-    conflicts
-        regular values: higher prio overrides
-        dictionaries: merge conflicting dictionaries
-        lists: higher prio overrides, merge any matching dictionary elements
-
-    assume target is low priority
     '''
     if isinstance(source, list) and isinstance(target, list):
         for i, value in enumerate(target):
-            if isinstance(value, dict) and isinstance(source[i], dict): # what if i not in source?
+            if i >= len(source):
```

Comment:
What if “i” is a string key of a dictionary ?

Reply:
This line is reached if both source and target are lists. If both are dictionaries, below block runs. If one of them is a dictionary and the other is not, no merge happens.

Reply:
Ah, yes you're right